zenoh/src/api/builders/publisher.rs

```diff
@@ -264,6 +267,19 @@ impl IntoFuture for PublicationBuilder<PublisherBuilder<'_, '_>, PublicationBuil
     }
 }
 
+/// Marks which [`PublisherBuilder`] QoS configurations were overwritten by Zenoh config.
+/// Associated [`PublisherBuilder`] methods calls will not modify overwritten configurations
+#[derive(Clone, Copy, Debug)]
+pub(crate) struct PublicationOverwrittenQoS {
```

Comment:
Maybe I'm missing something but this looks overcomplicated to me. 
Instead of overwriting values at builder creation and mark some values as overwritten with those booleans, why don't we simply overwrite the values at builder resolution ?

Reply:
Overwriting at builder resolution would require doing the lookup in `PublisherBuilder` resolution and in `SessionInner.resolve_put` since a `session.put` operation uses a `PublisherBuilder` to store and update the configuration. Doing the lookup in `resolve_put` means doing it for every publication made by a publisher, which would heavily impact the performance.

Reply:
To avoid doing the lookup in `resolve_put` we'd have to do it in every implementation of `Wait for PublicationBuilder` (currently 4)

Reply:
Done.